rdtools/test/filtering_test.py

```diff
@@ -1,96 +1,298 @@
+    assert ((expected_result == filtered).all())
+
+
+def test_logic_clip_filter(generate_power_time_series_no_clipping,
+                           generate_power_time_series_clipping,
+                           generate_power_time_series_one_min_intervals,
+                           generate_power_time_series_irregular_intervals):
+    ''' Unit tests for logic clipping filter.'''
+    power_no_datetime_index_nc, power_datetime_index_nc = \
+        generate_power_time_series_no_clipping
+    # Test that a Type Error is raised when a pandas series
+    # without a datetime index is used.
+    pytest.raises(TypeError,  logic_clip_filter,
+                  power_no_datetime_index_nc)
+    # Test that an error is thrown when we don't include the correct
+    # mounting configuration input
+    pytest.raises(ValueError,  logic_clip_filter,
+                  power_datetime_index_nc, 'not_fixed')
+    # Test that an error is thrown when there are 10 or fewer readings
+    # in the time series
+    pytest.raises(Exception,  logic_clip_filter,
+                  power_datetime_index_nc[:9])
+    # Generate 1-minute interval data, run it through the function, and
+    # check that the associated data returned is 1-minute
+    power_datetime_index_one_min_intervals = \
+        generate_power_time_series_one_min_intervals
+    mask_one_min = logic_clip_filter(power_datetime_index_one_min_intervals)
+    # Generate irregular interval data, and run it through the XGBoost model
+    power_datetime_index_irregular = \
+        generate_power_time_series_irregular_intervals
+    # Make sure that the routine throws a warning when the data sampling
+    # frequency is less than 95% consistent
+    warnings.simplefilter("always")
+    with warnings.catch_warnings(record=True) as w:
+        logic_clip_filter(power_datetime_index_irregular)
+        # Warning thrown for it being an experimental filter + irregular
+        # sampling frequency.
+        assert len(w) == 2
+    # Check that the returned time series index for the logic filter is
+    # the same as the passed time series index
+    mask_irregular = logic_clip_filter(power_datetime_index_irregular)
+    # Expect none of the sequence to be clipped (as it's
+    # constantly increasing)
+    mask_nc = logic_clip_filter(power_datetime_index_nc)
+    # Test the time series where the data is clipped
+    power_no_datetime_index_c, power_datetime_index_c = \
+        generate_power_time_series_clipping
+    # Expect 4 values in middle of sequence to be clipped (when x=50)
+    mask_c = logic_clip_filter(power_datetime_index_c)
+    filtered_c = power_datetime_index_c[mask_c]
+    assert bool(mask_nc.all(axis=None))
+    assert (len(filtered_c) == 96)
+    assert bool((mask_one_min.index.to_series().diff()[1:] ==
+                 np.timedelta64(60, 's')).all(axis=None))
+    assert bool((mask_irregular.index == power_datetime_index_irregular.index)
+                .all(axis=None))
```

Comment:
I think the calls to `bool` are unnecessary here, as well as `axis=None`.  FYI there's also [series.index.equals](https://pandas.pydata.org/docs/reference/api/pandas.Index.equals.html) for checking index equality